src/libs/gitlab/file_info.ts

```diff
@@ -29,21 +30,33 @@ const ensureRevisionsAreCloned = (files: Observable<FileInfo>): Observable<FileI
 /**
  * Resolves file information for a page with a single file, not including diffs with only one file.
  */
-export const resolveFileInfo = (codeView: HTMLElement): Observable<FileInfo> =>
-    of(undefined).pipe(
-        map(() => {
-            const { repoPath, filePath, rev } = getFilePageInfo()
-
-            return { repoPath, filePath, rev }
-        }),
-        filter(propertyIsDefined('filePath')),
-        switchMap(({ repoPath, rev, ...rest }) =>
-            resolveRev({ repoPath, rev }).pipe(
-                retryWhenCloneInProgressError(),
-                map(commitID => ({ ...rest, repoPath, commitID, rev: rev || commitID }))
+export const resolveFileInfo = (): Observable<FileInfo> => {
+    const { repoPath, filePath } = getFilePageInfo()
+    if (!filePath) {
+        return throwError(
+            new Error(
+                `Unable to determine the file path of the current file because the current URL (window.location ${
+                    window.location
+                }) does not have a file path.`
             )
         )
-    )
+    }
+    const commitID = commitIDFromPermalink({
+        selector: '.js-data-file-blob-permalink-url',
+        hrefRegex: new RegExp('^/.*?/.*?/blob/([0-9a-f]{40})/'),
+    })
+    if (isErrorLike(commitID)) {
+        return throwError(commitID)
+    }
+    // TypeScript doesn't realize that `commitID` is not an `Error` here.
+    const stringCommitID: any = commitID
```

Comment:
Same as the github file. 
```suggestion
    const stringCommitID = commitID as string
```